### PR TITLE
Interpolation cart search

### PR DESCRIPTION
**Pull request by vlipovac**
## Proposed changes

Implemented cartesian search for points in interpolation grid.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing (contribution related mainly to testing of existing functionality)
- [ ] Documentation update (if none of the other choices apply)
- [X] Other: Optimizing existing functionality

## Checklist

_Enter one &check; (`&check;`) in each row. The table can be updated after creating the PR._

||Yes|No|NA|
|---|---|---|---|
|The update is covered by the  test suite |_|_|&check;|
|The documentation is up-to-date |&check;|_|_|
|I have not duplicated existing functionality |&check;|_|_|

## Further comments

In future work, the interpolation and derivative approximation can be unified such that the point search has only to be performed once (contrary to as it is now, once for interpolation and once for each directional/axial derivative)

### CODE REVIEW

File: src/porepy/utils/interpolation_tables.py [hunk 1]
```diff
@@ -116,6 +121,15 @@ def interpolate(self, x: np.ndarray) -> np.ndarray:
 
         """
 
+        # allocate value array
+        values: np.ndarray
+        if len(x.shape) > 1:
+            values = np.zeros((self.dim, x.shape[1]))
+        else:
+            # in case a single point in the function domain was passed as a row vector
```
Comment (keileg) [inline src/porepy/utils/interpolation_tables.py:129]: Nice catch.

File: src/porepy/utils/interpolation_tables.py [hunk 2]
```diff
@@ -188,42 +205,33 @@ def diff(self, x: np.ndarray, axis: int) -> np.ndarray:
...
-        ind -= 1
-        # If a point is outside the interpolation grid, the argmax will return 0
-        # (either because the point is to the left, or it is to the right, which case
-        # the entire array to argmax is zero, thus the first value is 0)
-        if np.any(ind < 0):
-            raise ValueError(
-                f"Point outside coordinate range [{self._low}, {self._high}]"
-            )
+        ind = list()
+        # performing cartesian search per axis of the interpolation grid
```
Comment (keileg) [inline src/porepy/utils/interpolation_tables.py:224]:
```suggestion
        # performing Cartesian search per axis of the interpolation grid.
```

**Review by keileg — APPROVED** [review]
Very good. I found a typo, but that's it.

**Comment by IvarStefansson** [comment]
Why is testing NA?

**Pull request merged**